samples: bluetooth: mtu_update: Avoid busy looping & fix restarting advertising

manoj-aerlync opened the pull request:
Use semaphore to avoid busy-looping before BLE connection.
Check for NULL conn before retaking semaphore after disconnection

manoj-aerlync committed:
samples: bluetooth: mtu_update: peripheral: Avoid busy-looping

Avoid busy-looping before BLE connection and introduce
a semaphore. Check for NULL conn before retaking
semaphore after disconnection.

Signed-off-by: Manojkumar Konisetty <manoj@aerlync.com>

manoj-aerlync committed:
samples: bluetooth: mtu_update: peripheral: Restart Advertisement

Add a workqueue to handle readvertisement after
disconnection from central. This avoids performing
advertising operations directly from the disconnected callback,
ensuring better thread safety.

Signed-off-by: Manojkumar Konisetty <manoj@aerlync.com>

jhedberg commented:
Please include "samples" in the commit titles, since otherwise one gets the initial impression that this is related to the Bluetooth stack itself. I fixed the PR title already.

manoj-aerlync commented:
> Please include "samples" in the commit titles, since otherwise one gets the initial impression that this is related to the Bluetooth stack itself. I fixed the PR title already.

I have updated the commit title as suggested.Please review

jhedberg commented:
> I have updated the commit title as suggested.Please review

Thanks. Is the PR description correct? It talks about a NULL check, but I don't see that in the actual change.

jhedberg commented:
> > I have updated the commit title as suggested.Please review
> 
> Thanks. Is the PR description correct? It talks about a NULL check, but I don't see that in the actual change.

Nevermind. I can see it. I thought this was some leftover reference to https://github.com/zephyrproject-rtos/zephyr/pull/88889 which was already merged.
